tree/ntuple/v7/doc/architecture.md

```diff
@@ -3,19 +3,468 @@ RNTuple Code Architecture
+It is part of the `ROOTDataFrame` library.
+The RNTuple data source supports chains with a constructor that takes a list of input files.
+The RNTuple data source also supports multi-threaded dataframes, parallelized on the file and cluster level.
+
+The data source exposes inner fields of complex collections.
+For instance, if the data model contains a vector of `Event` classes, where each `Event` has `pt` and `eta` floats,
+the dataframe can use the event vector itself (`Event` column) as well as the `float` columns `Event.pt` and `Event.eta`.
+
+### RClusterPool
+The RClusterPool is an internal class owned be a page source.
+The cluster pool maintains an I/O thread that asynchronously prefetches the next few clusters.
+Through `RPageSource::SetEntryRange()`, the cluster pool is instructed to not read beyond the given limit.
+This is used in the RNTuple data source when multiple threads work on different clusters of the same file.
+
+### RMiniFile
+The RMiniFile is an internal class used to read and write RNTuple data in a ROOT file.
+It provides a minimal subset of the `TFile` functionality.
+Its purpose is to reduce the coupling between RNTuple and the ROOT I/O library.
+
+For writing data, the RMiniFile can either use a proper `TFile` (descendant) or a C file stream (only for new ROOT files with a single RNTuple).
+For reading, the `RMiniFile` always uses an `RRawFile`.
+
+### RRawFile
+The RRawFile internal abstract class provides an interface to read byte ranges from a file, including vector reads.
+Concrete implementations exist for local files, XRootD and HTTP (the latter two through the ROOT plugin mechanism).
+The local file implementation on Linux uses uring for vector reads, if available.
+A raw file that uses a TFile (to be implemented) provides access to the full set of TFile implementations, e.g. TMemFile.
+
+Tooling
+-------
+
+### RNTupleMerger
+The `RNTupleMerger` is an internal class and part of the core RNTuple library.
+It concatenates RNTuple data from several sources into a combined sink.
+It implements "fast merging", i.e. copy-based merging that does not decompress and recompress pages.
+The RNTupler merger is used by the `TFileMerger` and thus provides RNTuple merge support in `hadd` and `TBufferMerger`.
+
+### RNTupleImporter
+The RNTupleImporter creates RNTuple data sets from ROOT trees.
+It is part of the `ROOTNTupleUtil` library.
+
+### RNTupleInspector
+The RNTupleInspector provides insights of an RNTuple, e.g. the distribution of data volume wrt. column types.
+It is part of the `ROOTNTupleUtil` library.
+
+Ownership Model
+---------------
+
+By default, objects involved in RNTuple I/O (objects read from disk or written to disk) are passed to RNTuple as shared pointers.
+Both RNTuple or the application may create the object.
+Raw pointers to objects can be passed to RNTuple -- such objects are considered as owned by the application with a sufficiently long lifetime.
```

Comment:
Maybe be more explicit and add that is the caller's responsibility to guarantee the lifetime of the objects